src/auth/service/auth.service.spec.ts

```diff
@@ -0,0 +1,79 @@
+import { Test, TestingModule } from '@nestjs/testing'
+import { HttpException } from '@nestjs/common'
+import { AuthService } from './auth.service'
+import { UserService } from '../../user/service/user.service'
+import { getRepositoryToken } from '@nestjs/typeorm'
+import { StudentEntity } from '../../students/entities/students.entity'
+import { JwtModule } from '@nestjs/jwt'
+import { jwtConstants } from '../constants'
+import { CreateStudentDTO } from '../../students/model/student.dto.input'
+import { NotFoundException } from '@nestjs/common'
+
+describe('User Login Service', () => {
+  let authService: AuthService
+  let userService: UserService
+
+  const mockRepository = {
+    find: jest.fn(),
+    findOneBy: jest.fn(),
+    create: jest.fn()
+  }
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      providers: [
+        UserService,
+        AuthService,
+        {
+          provide: getRepositoryToken(StudentEntity),
+          useValue: mockRepository
+        }
+      ],
+      imports: [
+        JwtModule.register({
+          secret: jwtConstants.secretKey,
+          signOptions: { expiresIn: jwtConstants.expirationTime }
+        })
+      ]
+    }).compile()
+    authService = await moduleFixture.resolve(AuthService)
+    userService = await moduleFixture.resolve(UserService)
+  })
+
+  beforeEach(() => {
+    mockRepository.find.mockReset()
+    mockRepository.findOneBy.mockReset()
+    mockRepository.create.mockReset()
+  })
+
+  describe('Login Test', () => {
+    it('Should return NotFoundException', async () => {
+      const tax_id = '123131'
+      await expect(userService.findUserByTaxId(tax_id)).rejects.toThrow(
+        NotFoundException
+      )
+    })
+  })
+
+  it('Should return UnauthorizedException', async () => {
+    const student = new CreateStudentDTO(
```

Comment:
Caraio, não levei fé que ia meter os testes kkkk, top!
Só um feedback bobo, tem uma parte de TestUtils em algum lugar do código, onde você pode criar esses init da classe, pra não ficar muito grande seus testes, daí quando for fazer os próximos você experimenta colocar lá por exemplo
TestUtils.ReturnValidStudentDTO, dai dentro do método você designa as propriedades dele